src/middleware/validate/validateBooleanMiddleware.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright 2017-2022 Parity Technologies (UK) Ltd.
+// This file is part of Substrate API Sidecar.
+//
+// Substrate API Sidecar is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import { RequestHandler } from 'express';
+import { BadRequest } from 'http-errors';
+
+// Query Params used in controllers
+const queryParams = [
+	'eventDocs', // BlocksController
+	'extrinsicDocs', // BlocksController
+	'finalized', // BlocksController
+	'denominated', // AccountsBalanceController
+	'unclaimedOnly', // AccountsStakingPayoutsController
+	'actions', // BlocksTraceController
+	'includeFee', // NodeTransactionPoolController
+	'adjustMetadataV13', // PalletsStorageController
+	'currentLeaseHolders', // ParasController
+	'noMeta', // TransactionMaterialController
+];
+
+export const validateBooleanMiddleware: RequestHandler = (req, _res, next) => {
+	const queryKeys = Object.keys(req.query);
+	const errQueryParams: string[] = [];
+	queryKeys
+		.filter((queryParam) => queryParams.includes(queryParam))
+		.forEach((queryParam) => {
+			const queryParamVal =
+				typeof req.query[queryParam] === 'string'
+					? (req.query[queryParam] as string).toLowerCase()
+					: '';
+			if (!(queryParamVal === 'true' || queryParamVal === 'false')) {
```

Comment:
I guess this is the idiomatic way of checking the queryParams.

Reply:
I guess it's because all query params are strings.

Maybe worth doing `queryParamVal.toLowerCase()` to support eg TRUE or True? Not too bothered though; nothing wrong with being strict and accepting only `true` or `false`.

In some apps, the mere presence or absense of a query param sets it to true or false, eg `?foo&bar` leads to foo and bar being true. Again, I'm happy if we just want to be stirct and support only `=true` or `=false`; just throwing it out there :)

Reply:
> Maybe worth doing queryParamVal.toLowerCase() to support eg TRUE or True? Not too bothered though; nothing wrong with being strict and accepting only true or false.

I actually already set `queryParamVal` to lower case above.

Reply:
Oh yeah; apologies; I somehow missed that entirely :)